ee/codegen/src/types/vellum.ts

```diff
@@ -660,17 +660,22 @@ export interface NodeOutput {
   value?: WorkflowValueDescriptor;
 }
 
+export interface GenericNodeData {
+  label: string;
+  trigger: NodeTrigger;
+  ports: NodePort[];
+  adornments?: AdornmentNode[];
+  attributes: NodeAttribute[];
+  outputs: NodeOutput[];
+}
+
 export interface GenericNode extends BaseWorkflowNode {
   id: string;
   type: "GENERIC";
+  data: GenericNodeData;
   displayData?: GenericNodeDisplayData;
   base: CodeResourceDefinition;
   definition?: CodeResourceDefinition;
-  trigger: NodeTrigger;
-  ports: NodePort[];
-  adornments?: AdornmentNode[];
-  attributes: NodeAttribute[];
-  outputs: NodeOutput[];
```

Comment:
to add, the long term goal is actually to go the opposite direction. Other Workflow Nodes -> Generic Nodes